src/opm/input/eclipse/share/keywords/000_Eclipse100/W/WCONPROD

```diff
@@ -18,7 +18,8 @@
     {
       "item": 3,
       "name": "CMODE",
-      "value_type": "STRING"
+      "value_type": "STRING",
+      "default": "NONE"
```

Comment:
Hmm, this is an interesting side effect with actionx for the failing test ACTIONX_GCONPROD. If I remove the default the test runs rather fast and succeeds. But with the default it takes forever and fails. I guess I need to dig into this one... No idea what actually happens.

Reply:
Arrgh might have run the wrong executable for the check.

Reply:
Nevertheless this is the change that causes the failire and substantially increases the runtime in debug mode,

Reply:
i cannot see it increasing any runtime, but i do see this
```
==1840963== Conditional jump or move depends on uninitialised value(s)
==1840963==    at 0x239426: int const& std::max<int>(int const&, int const&) (stl_algobase.h:227)
==1840963==    by 0x30C7795: (anonymous namespace)::Evaluator::FunctionRelation::update(unsigned long, double, (anonymous namespace)::Evaluator::InputData const&, (anonymous namespace)::Evaluator::SimulatorResults const&, Opm::SummaryState&) const (Summary.cpp:2650)
==1840963==    by 0x30CD3A9: Opm::out::Summary::SummaryImplementation::eval(int, double, Opm::data::Wells const&, Opm::data::GroupAndNetworkValues const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, double> > >&, Opm::Inplace const&, Opm::Inplace const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<double, std::allocator<double> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<double, std::allocator<double> > > > > const&, std::map<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>, double, std::less<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> >, std::allocator<std::pair<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> const, double> > > const&, std::map<int, Opm::data::AquiferData, std::less<int>, std::allocator<std::pair<int const, Opm::data::AquiferData> > > const&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Opm::data::InterRegFlowMap, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Opm::data::InterRegFlowMap> > > const&, Opm::SummaryState&) const (Summary.cpp:3908)
==1840963==    by 0x30D109D: Opm::out::Summary::eval(Opm::SummaryState&, int, double, Opm::data::Wells const&, Opm::data::GroupAndNetworkValues const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, double> > >, Opm::Inplace const&, Opm::Inplace const&, Opm::PAvgCalculatorCollection const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<double, std::allocator<double> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<double, std::allocator<double> > > > > const&, std::map<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>, double, std::less<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> >, std::allocator<std::pair<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> const, double> > > const&, std::map<int, Opm::data::AquiferData, std::less<int>, std::allocator<std::pair<int const, Opm::data::AquiferData> > > const&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, Opm::data::InterRegFlowMap, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, Opm::data::InterRegFlowMap> > > const&) const (Summary.cpp:4352)
==1840963==    by 0x1D45E63: Opm::EclGenericWriter<Dune::CpGrid, Dune::CpGrid, Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid> >, Dune::MultipleCodimMultipleGeomTypeMapper<Dune::GridView<Dune::DefaultLeafGridViewTraits<Dune::CpGrid> >, Dune::Impl::MCMGFailLayout>, double>::evalSummary(int, double, std::map<unsigned long, double, std::less<unsigned long>, std::allocator<std::pair<unsigned long const, double> > > const&, Opm::data::Wells const&, Opm::data::GroupAndNetworkValues const&, std::map<int, Opm::data::AquiferData, std::less<int>, std::allocator<std::pair<int const, Opm::data::AquiferData> > > const&, std::map<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int>, double, std::less<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> >, std::allocator<std::pair<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int> const, double> > > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, double, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, double> > > const&, std::map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::vector<double, std::allocator<double> >, std::less<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, std::vector<double, std::allocator<double> > > > > const&, Opm::Inplace const&, Opm::Inplace const&, Opm::EclInterRegFlowMap const&, Opm::SummaryState&, Opm::UDQState&) (eclgenericwriter.cc:512)
==1840963==    by 0x33700B: Opm::EclWriter<Opm::Properties::TTag::EclFlowProblem>::evalSummaryState(bool) (eclwriter.hh:258)
==1840963==    by 0x30458A: Opm::EclProblem<Opm::Properties::TTag::EclFlowProblem>::endTimeStep() (eclproblem.hh:1159)
==1840963==    by 0x2DEC58: Opm::BlackoilModelEbos<Opm::Properties::TTag::EclFlowProblem>::afterStep(Opm::SimulatorTimerInterface const&) (BlackoilModelEbos.hpp:429)
==1840963==    by 0x2BB8B5: Opm::NonlinearSolverEbos<Opm::Properties::TTag::EclFlowProblem, Opm::BlackoilModelEbos<Opm::Properties::TTag::EclFlowProblem> >::step(Opm::SimulatorTimerInterface const&) (NonlinearSolverEbos.hpp:228)
==1840963==    by 0x2B98F8: Opm::SimulatorReport Opm::AdaptiveTimeSteppingEbos<Opm::Properties::TTag::EclFlowProblem>::step<Opm::NonlinearSolverEbos<Opm::Properties::TTag::EclFlowProblem, Opm::BlackoilModelEbos<Opm::Properties::TTag::EclFlowProblem> > >(Opm::SimulatorTimer const&, Opm::NonlinearSolverEbos<Opm::Properties::TTag::EclFlowProblem, Opm::BlackoilModelEbos<Opm::Properties::TTag::EclFlowProblem> >&, bool, std::vector<int, std::allocator<int> > const*) (AdaptiveTimeSteppingEbos.hpp:413)
==1840963==    by 0x29792D: Opm::SimulatorFullyImplicitBlackoilEbos<Opm::Properties::TTag::EclFlowProblem>::runStep(Opm::SimulatorTimer&) (SimulatorFullyImplicitBlackoilEbos.hpp:260)
==1840963==    by 0x27E584: Opm::SimulatorFullyImplicitBlackoilEbos<Opm::Properties::TTag::EclFlowProblem>::run(Opm::SimulatorTimer&) (SimulatorFullyImplicitBlackoilEbos.hpp:153)
==1840963== 
```
@bska i guess you might be knowledgable here. SummaryNode::number is uninitialized and seems to miss an assignment here. i'm not sure what is the correct fix.

Reply:
> SummaryNode::number is uninitialized and seems to miss an assignment here. i'm not sure what is the correct fix.

We only supposed to read from `SummaryNode::number` if the corresponding summary vector applies to

* An aquifer, e.g., `AAQT` or `ANQT`
* A cell (block), e.g.,`BOPR`
* A well's reservoir connection, e.g., `COPT`
* A region or a pair of regions, e.g., `ROPT` or `ROFR+`
* A well segment, e.g., `SOFR`

The `::number` is essentially meaningless for other categories and it's arguably an error to look at its value in that case.  Without knowing **which** summary vector is being evaluated here, and in what context, it's hard to say what the correct course of action is.  That said, I'd be reluctant to add a default/dummy value without a good justification.